### ISSUE TITLE
Civic card Share button needs to either be a button or support spacebar activation

### ISSUE THREAD

**Issue by fstorr** [hackoregon/civic#871]
## what

The [Civic card](https://hackoregon.github.io/civic/?path=/story/component-lib-story-cards-civic-card--layout-classic) Share "button" has this markup:

```
<a tabindex="0" role="button"><i class="fa fa-link"></i><span>Share</span></a>
```

The ARIA `role="button"` attribute tells assistive technology (e.g. a screen reader) that the element is a button. What it doesn't do is to actually make it behave like a button and work when the space bar is pressed (links are keyboard activated with the Return key, and buttons by the Return key or the space bar).

## How to fix:

Do one of these:

1. Keep the existing code but add JavaScript support for the space bar so runs the function in the same way the Return key does.
2. Replace the existing code with a `<button>` element, which is the correct one for this component:

```
<button type="button">
  <i aria-hidden="true" class="fa fa-link"></i><span>Share</span>
</button>
```

Also note that the icon has the `aria-hidden="true"` attribute, which hides it from assistive technology. This should be added to the link option if that option is used instead of the button.

**Comment by jaronheard** [comment]
#1084 addressed
